json_annotation/lib/src/json_key.dart

```diff
@@ -136,3 +136,5 @@ class JsonKey {
 }
 
 enum _NullAsDefault { value }
+
+enum IncludeWith { both, toJson, fromJson, ignore, legacy }
```

Comment:
I added a fifth option, currently called `legacy`. Fields that use this option should behave as before, so currently this is the default option.